lib/services/access_request.go

```diff
@@ -2403,7 +2412,15 @@ func (m *RequestValidator) pruneResourceRequestRoles(
 				`no roles configured in the "search_as_roles" for this user allow `+
 					`access to at least one requested resources. `+
 					`resources: %s roles: %v unmatched resources: %v`,
-				resourcesStr, roles, kubeResourceMatcher.Unmatched())
+				resourcesStr, requestedRoles, kubeResourceMatcher.Unmatched())
+		}
+		if resource.GetKind() == types.KindSAMLIdPServiceProvider {
```

Comment:
Arguably abusing `pruneResourceRequestRoles` method to do many things but this is the place where roles and resource are listed and looped through and seemed like best place to slip this block.